tests/embind/test_val.cpp

```diff
@@ -593,6 +602,18 @@ int main()
   ensure(aAsArray.at(2).as<string>() == "b");
   ensure(aAsArray.size() == 4);
   
+  test("template<typename T> std::vector<T *> vecFromJSArray(const val& v)");
+  EM_ASM(
+    b = [];
+    b[0] = Module.makeDummy();
+    b[1] = Module.makeDummy();
+  );
+  const std::vector<Dummy *>& bAsArray = vecFromJSArray<Dummy *>(val::global("b"), allow_raw_pointers());
+  ensure(bAsArray.size() == 2);
+  for (auto *dummy : bAsArray) {
+    delete dummy;
```

Comment:
FWIW you could instead just create `static Dummy` and return const pointers to those, rather than dealing with heap-allocation and deallocation. But I don't have a strong preference either way.